### ISSUE TITLE
Half of the asset is grayscaled

### ISSUE THREAD

**Issue by osrl** [davemorrissey/subsampling-scale-image-view#52]
Half of the image is decoded grayscale. I'm on 5.0.1 (didn't test the rest). I'll upload the file here maybe you can reproduce the issue. I don't know if github process the image so I'll upload to [dropbox](https://www.dropbox.com/s/kzikoeduf1x9i2e/map.jpg?dl=0) too. 

![gsa_map](https://cloud.githubusercontent.com/assets/781453/6641382/1d00c6f2-c9a6-11e4-8667-61d01f8ca83e.jpg)

**Comment by davemorrissey** [comment]
This is very strange, I've never seen anything like it. For me, I get an entirely grey image when zoomed out, but when I zoom in I get coloured tiles that I can only describe as "1970s TV quality". Here's what I know so far:

* Changing the decoding to ARGB_8888, enabling dithering and other config changes make no difference at all.
* Using the RapidImageRegionDecoder class, which is based on a different JPG decoder library, gives me a properly decoded full colour image when zoomed out but occasionally loads tile in greyscale when zoomed in.
* ImageMagick `identify` shows nothing unusual about the image format compared to my test images.
* Re-saving the image in GIMP fixes it. The only obvious difference in the output from `identify` is that the fixed image has a resolution attribute.

I know the Skia image library is full of bugs and the reports I've submitted have been ignored, but given an entirely different jpg decoder library also fails to handle this image, I can only suggest you re-save it from another graphics package.

**Comment by osrl** [comment]
Ok thank you. I'll try that. Of course this is a temporary solution but that fixes it for me.

**Comment by laurencedawson** [comment]
I've had this reported for a few more images in Sync:

https://i.imgur.com/X9f3KtD.jpg

Aside from re-encoding is there another workaround? Cheers

**Comment by davemorrissey** [comment]
The changes I made some time ago where the view will load the whole bitmap with BitmapFactory if it's small enough, avoiding BitmapRegionDecoder's bugs, was the last of my box of tricks. These bugs have been fixed in the Skia project but I expect Nougat will be the first version of Android to get the update and I don't know whether it will be backported - and even then many older phones won't get updated.

The last option was attempting to build and embed a Skia native library from the most recent code, but there would be one for each architecture and I think it was about 50Mb (I think we've discussed this before). That's far beyond my experience and I can't afford the time to learn it.
